message_ix_models/model/material/build.py

```diff
@@ -65,98 +68,55 @@
-    # ixmp://ene-ixmp/CD_Links_SSP2_v2.1_clean/baseline
-    if {"World", "R11_GLB"} < set(info.set["node"]):
-        log.warning("Remove 'R11_GLB' from node list for data generation")
-        info.set["node"].remove("R11_GLB")
-    if {"World", "R12_GLB"} < set(info.set["node"]):
-        log.warning("Remove 'R12_GLB' from node list for data generation")
-        info.set["node"].remove("R12_GLB")
-
-    for func in DATA_FUNCTIONS_1 + DATA_FUNCTIONS_2:
+    for func in DATA_FUNCTIONS:
         # Generate or load the data; add to the Scenario
         log.info(f"from {func.__name__}()")
         data = func(scenario)
-        # if "SSP_dev" in scenario.model:
-        #     if "emission_factor" in list(data.keys()):
-        #         data.pop("emission_factor")
+        data = {k: v for k, v in data.items() if not v.empty}
         add_par_data(scenario, data, dry_run=dry_run)
-
-    log.info("done")
-
-
-def add_data_2(scenario, dry_run=False):
-    """Populate `scenario` with MESSAGEix-Materials data."""
-    # Information about `scenario`
-    info = ScenarioInfo(scenario)
-
-    # Check for two "node" values for global data, e.g. in
-    # ixmp://ene-ixmp/CD_Links_SSP2_v2.1_clean/baseline
-    if {"World", "R11_GLB"} < set(info.set["node"]):
-        log.warning("Remove 'R11_GLB' from node list for data generation")
-        info.set["node"].remove("R11_GLB")
-    if {"World", "R12_GLB"} < set(info.set["node"]):
-        log.warning("Remove 'R12_GLB' from node list for data generation")
-        info.set["node"].remove("R12_GLB")
-
-    for func in DATA_FUNCTIONS_2:
-        # Generate or load the data; add to the Scenario
-        log.info(f"from {func.__name__}()")
-        # TODO: remove this once emission_factors are back in SSP_dev
-        data = func(scenario)
-        # if "SSP_dev" in scenario.model:
-        #     if "emission_factor" in list(data.keys()):
-        #         data.pop("emission_factor")
-        add_par_data(scenario, data, dry_run=dry_run)
-
     log.info("done")
 
 
 def build(
-    scenario: message_ix.Scenario, old_calib: bool, iea_data_path=None
+    context: Context,
+    scenario: message_ix.Scenario,
+    old_calib: bool,
+    modify_existing_constraints=True,
+    iea_data_path=None,
 ) -> message_ix.Scenario:
     """Set up materials accounting on `scenario`."""
+    node_suffix = context.model.regions
 
-    # Get the specification
-    # Apply to the base scenario
-    spec = get_spec()
-
-    if "water_supply" not in list(scenario.set("level")):
-        scenario.check_out()
-        # add missing water tecs
-        scenario.add_set("technology", "extract__freshwater_supply")
-        scenario.add_set("level", "water_supply")
-        scenario.add_set("commodity", "freshwater_supply")
-
-        water_dict = pd.read_excel(
-            package_data_path("material", "other", "water_tec_pars.xlsx"),
-            sheet_name=None,
+    if node_suffix != "R12":
+        raise NotImplementedError(
```

Comment:
This line is not covered by tests, but if it should be, that's quite simple: call the function with a context that has `context.model.region = "R11"`, for example, and tell it it should expect a `NotImplementedError`.